### PR TITLE
Add parameter start-after to ListObjectsV2

### PR DESCRIPTION
**Pull request by sinhaashish**
Add parameter start-after to ListObjectsV2 to make the iteration uninterruptible. 

Resolves the [issue](https://github.com/minio/minio-js/issues/733)

### CODE REVIEW

**Comment by ebozduman** [comment]
It'd be a good idea to visit and modify the `functional-tests.js` file to add the new argument, `startAfter`, although existing `listObjectsV2` test still runs fine, since `startAfter` is optional.
However, it is preferably better to add a new test to check the new functionality.

File: src/main/minio.js [hunk 1]
```diff
@@ -1239,7 +1239,7 @@ export class Client {
   }
 
   // list a batch of objects using S3 ListObjects v2
-  listObjectsV2Query(bucketName, prefix, continuationToken, delimiter, maxKeys) {
+  listObjectsV2Query(bucketName, prefix, continuationToken, delimiter, maxKeys,startAfter) {
```
[the remark this answers is not in the document]
Comment (sinhaashish) [reply src/main/minio.js]: Modified.

File: src/main/minio.js [hunk 2]
```diff
@@ -1301,17 +1309,18 @@ export class Client {
   // * `bucketName` _string_: name of the bucket
   // * `prefix` _string_: the prefix of the objects that should be listed (optional, default `''`)
   // * `recursive` _bool_: `true` indicates recursive style listing and `false` indicates directory style listing delimited by '/'. (optional, default `false`)
-  //
```
[the remark this answers is not in the document]
Comment (sinhaashish) [reply src/main/minio.js]: Done

File: src/main/minio.js [hunk 3]
```diff
@@ -1239,7 +1239,7 @@ export class Client {
   }
 
   // list a batch of objects using S3 ListObjects v2
```
[the remark this answers is not in the document]
Comment (sinhaashish) [reply src/main/minio.js]: Modified

File: docs/API.md [hunk 4]
```diff
@@ -259,6 +259,7 @@ __Parameters__
 | `bucketName` | _string_ | Name of the bucket. |
 | `prefix`  | _string_  |  The prefix of the objects that should be listed (optional, default `''`). |
 | `recursive`  | _bool_  | `true` indicates recursive style listing and `false` indicates directory style listing delimited by '/'. (optional, default `false`).  |
+| `startAfter`  | _string_  |  Specifies the key to start after when listing objects in a bucket. (optional, default `''`). |
```
[the remark this answers is not in the document]
Comment (sinhaashish) [reply docs/API.md]: Modified

**Comment by sinhaashish** [comment]
> It'd be a good idea to visit and modify the `functional-tests.js` file to add the new argument, `startAfter`, although existing `listObjectsV2` test still runs fine, since `startAfter` is optional.
> However, it is preferably better to add a new test to check the new functionality.

Added the filed with blank value for start-After.

**Comment by sinhaashish** [comment]
@ebozduman   PTAL

File: src/main/minio.js [hunk 5]
```diff
@@ -1238,8 +1238,17 @@ export class Client {
...
   }
 
-  // list a batch of objects using S3 ListObjects v2
-  listObjectsV2Query(bucketName, prefix, continuationToken, delimiter, maxKeys) {
+  // listObjectsV2Query - (List Objects V2) - List some or all (up to 1000) of the objects in a bucket.
+  //
+  // You can use the request parameters as selection criteria to return a subset of the objects in a bucket.
+  // request parameters :-
+  // ---------
+  // ?prefix - Limits the response to keys that begin with the specified prefix.
```
[the remark this answers is not in the document]
Comment (sinhaashish) [reply src/main/minio.js]: Removed. PTAL

**Pull request merged**